packages/runtime/src/builder/serialization/control-serialization.ts

```diff
@@ -12,90 +11,121 @@ import {
   RichTextDescriptor as RichTextControl,
-  RichTextValue as RichTextControlValue,
   ShapeDescriptor as ShapeControl,
-  ShapeValue as ShapeControlValue,
   TextInputDescriptor as TextInputControl,
-  TextInputValue as TextInputControlValue,
   TextStyleDescriptor as TextStyleControl,
-  TextStyleValue as TextStyleControlValue,
   TypeaheadDescriptor as TypeaheadControl,
-  TypeaheadOptions as TypeaheadControlConfig,
   TypeaheadValue as TypeaheadControlValue,
   PanelDescriptor as PanelControl,
-  PanelDescriptorType as PanelControlType,
   PanelDescriptorValueType as PanelControlValueType,
   PropControllerDescriptor as Control,
   Props as Controls,
 } from '../../prop-controllers'
-import {
-  IconRadioGroupOption,
-  LengthOption,
-  SelectLabelOrientation,
-  SelectOption,
-  Length,
-} from '../../prop-controllers/descriptors'
 import {
   deserializeComboboxControlDefinition,
   serializeComboboxControlDefinition,
 } from './controls/combobox'
-import { deserializeListControlDefinition, serializeListControlDefinition } from './controls/list'
-import { deserializeRichTextControlV2, serializeRichTextControlV2 } from './controls/rich-text-v2'
-import {
-  deserializeShapeControlDefinition,
-  serializeShapeControlDefinition,
-} from './controls/shape'
-import { deserializeStyleV2Control, serializeStyleV2Control } from './controls/style-v2'
 import { Deserialize, Serialize } from './controls/types'
 import {
-  DeserializedFunction,
   deserializeFunction,
   isSerializedFunction,
-  SerializedFunction,
   serializeFunction,
 } from './function-serialization'
+import {
+  DeserializedCheckboxControl,
+  DeserializedControl,
+  DeserializedDateControl,
+  DeserializedGapXControl,
+  DeserializedGapYControl,
+  DeserializedImageControl,
+  DeserializedLinkControl,
+  DeserializedListControl,
+  DeserializedNumberControl,
+  DeserializedPanelControl,
+  DeserializedResponsiveColorControl,
+  DeserializedResponsiveIconRadioGroupControl,
+  DeserializedResponsiveLengthControl,
+  DeserializedResponsiveNumberControl,
+  DeserializedResponsiveSelectControl,
+  DeserializedRichTextControl,
+  DeserializedShapeControl,
+  DeserializedTextInputControl,
+  DeserializedTextStyleControl,
+  DeserializedTypeaheadControl,
+  SerializedCheckboxControl,
+  SerializedControl,
+  SerializedDateControl,
+  SerializedGapXControl,
+  SerializedGapYControl,
+  SerializedImageControl,
+  SerializedLinkControl,
+  SerializedListControl,
+  SerializedNumberControl,
+  SerializedPanelControl,
+  SerializedPanelControlValueType,
+  SerializedResponsiveColorControl,
+  SerializedResponsiveIconRadioGroupControl,
+  SerializedResponsiveLengthControl,
+  SerializedResponsiveNumberControl,
+  SerializedResponsiveSelectControl,
+  SerializedRichTextControl,
+  SerializedShapeControl,
+  SerializedTextInputControl,
+  SerializedTextStyleControl,
+  SerializedTypeaheadControl,
+} from './types'
+
+function serializeShapeControlDefinition(
```

Comment:
This is one the big changes: I ended up moving a bunch of the modules that are recursively dependent into the same file. It sort of sucks that we can't have these be their own modules but this is sort of the best approach to avoid these circular deps. I'm open to recommendations on how to split these up even better, like splitting this up into `serialization` and `deserialization` modules 🙏 

cc. @migueloller @joshuawootonn 

Here is also a [SO article](https://stackoverflow.com/questions/63414806/how-to-avoid-circular-dependency-between-modules-when-working-with-recursive-fun) that I found useful here.